src/backend/distributed/ddl/foreign_constraint.c

```diff
@@ -309,6 +369,94 @@ ErrorIfUnsupportedForeignConstraint(Relation relation, char distributionMethod,
 }
 
 
+bool
+ForeignKeyExistsToReferenceTableOnColumn(char *affectedColumnName,
+										 Oid relationId)
+{
+	Relation pgConstraint = NULL;
+	SysScanDesc scanDescriptor = NULL;
+	ScanKeyData scanKey[1];
+	int scanKeyCount = 1;
+	HeapTuple heapTuple = NULL;
+	bool foreignKeyToReferenceTableIncludesGivenColumn = false;
+	bool isNull = false;
+	Datum columnsDatum;
+	Datum *columnArray;
+	int columnCount = 0;
+	int attrIdx;
+
+	pgConstraint = heap_open(ConstraintRelationId, AccessShareLock);
+
+	ScanKeyInit(&scanKey[0],
+				Anum_pg_constraint_connamespace,
+				BTEqualStrategyNumber, F_OIDEQ,
+				get_rel_namespace(relationId));
+
+	scanDescriptor = systable_beginscan(pgConstraint, ConstraintNameNspIndexId, true,
```

Comment:
We're not providing the constraint name to the index scan. This seems to have some weird effects (probably wrong results or assertion failures). 

Here it seems safer to use `contype` as the index and skip unrelated relations in the loop below.